src/dodal/devices/i11/i11_robot.py

```diff
@@ -0,0 +1,149 @@
+        self.servo_on = epics_signal_rw(bool, prefix + "SVON")  # Servo on/off
+        self.err = epics_signal_rw(int, prefix + "ERR")
+
+        self.robot_sample_state = epics_signal_rw(float, prefix + "D010")
+        self.next_sample_position = epics_signal_rw_rbv(
+            int, prefix + "D011", read_suffix=":RBV"
+        )
+        self.current_sample_position = epics_signal_rw_rbv(
+            int, prefix + "D012", read_suffix=":RBV"
+        )
+        self.door_latch_state = epics_signal_rw(int, prefix + "NEEDRECOVER")
+
+        super().__init__(name=name)
+
+    async def recover(self):
+        await asyncio.gather(
+            self.start.trigger(),
+            set_and_wait_for_value(self.job, RobotJobs.RECOVER),
+            set_and_wait_for_value(self.err, False),
+        )
+
+    async def clear_sample(self, table_in=True):
+        sample_state = await self.robot_sample_state.get_value()
+        if sample_state == RobotSampleState.DIFF:
+            await asyncio.gather(
+                set_and_wait_for_value(self.job, RobotJobs.PICKD),
+                set_and_wait_for_value(self.job, RobotJobs.PLACEC),
+            )
+        elif sample_state == RobotSampleState.ONGRIP:
+            await set_and_wait_for_value(self.job, RobotJobs.PLACEC)
+        elif sample_state == RobotSampleState.CAROSEL:
+            pass
+        elif sample_state == RobotSampleState.UNKNOWN:
+            LOGGER.error("UNKNOWN sample state from robot, exit")
+
+        if table_in:
+            await set_and_wait_for_value(self.job, RobotJobs.TABLEIN)
+
+        LOGGER.info("Sample cleared from diffractometer")
+
+    async def start_robot(self):
+        await asyncio.gather(
+            set_and_wait_for_value(self.servo_on, True),
+            set_and_wait_for_value(self.hold, False),
+            self.start.trigger(),
+        )
+
+    async def load_sample(self, sample_location: int):
+        sample_state = await self.robot_sample_state.get_value()
+        if sample_state == RobotSampleState.CAROSEL:
+            await set_and_wait_for_value(self.job, RobotJobs.PICKC)
+            await set_and_wait_for_value(self.job, RobotJobs.PLACED)
+        elif sample_state == RobotSampleState.ONGRIP:
+            await set_and_wait_for_value(self.job, RobotJobs.PLACED)
+        elif sample_state == RobotSampleState.DIFF:
+            pass
+        elif sample_state == RobotSampleState.UNKNOWN:
+            LOGGER.warning(f"No sample at sample holder position {sample_location}")
+            LOGGER.error("UNKNOWN sample state from robot, exit")
+
+    @AsyncStatus.wrap
+    async def set(self, sample_location: int) -> None:
+        if not (
+            self.MIN_NUMBER_OF_SAMPLES <= sample_location <= self.MAX_NUMBER_OF_SAMPLES
+        ):
+            raise ValueError(
+                f"Sample location must be between {self.MIN_NUMBER_OF_SAMPLES} and {self.MAX_NUMBER_OF_SAMPLES}, got {sample_location}"
+            )
```

Comment:
```suggestion
        if sample_location < self.MIN_NUMBER_OF_SAMPLES or sample_location > self.MAX_NUMBER_OF_SAMPLES
        ):
            raise ValueError(
                f"Sample location must be between {self.MIN_NUMBER_OF_SAMPLES} and {self.MAX_NUMBER_OF_SAMPLES}, got {sample_location}"
            )
```

nit: I think it's slightly clearer to read like this- I was about to try and adjust to `MIN_NUMBER_OF_SAMPLES < sample_location` because I didn't see the preceding `not`